### PR TITLE
Support no image size in PascalVOC loader

### PR DESCRIPTION
**Pull request by LukeWood**
This will allow us to support Ragged pipelines where the batch sizes are randomly set each batch.

### CODE REVIEW

**Review by tanzhenyu — COMMENTED** [review]
According to the description, "where the batch sizes" -- is batch sizes related to the image sizes in this PR?

**Comment by LukeWood** [comment]
TODO:

- update keras.io guides to specify image size
- make sure other places using the data loader don't rely on img_size

**Comment by tanzhenyu** [comment]
> TODO:
> 
> * update keras.io guides to specify image size
> * make sure other places using the data loader don't rely on img_size

Related question -- any chance we don't make this as a public API? This seems to be a very thin wrapper for now and also not very stable.

**Comment by LukeWood** [comment]
You can't batch non-resized images, so I've removed the default batch size to make the default constructor work.

**Comment by tanzhenyu** [comment]
> You can't batch non-resized images, so I've removed the default batch size to make the default constructor work.

Oh you submitted another commit after my initial review.

File: keras_cv/datasets/pascal_voc/load.py [hunk 1]
```diff
@@ -69,10 +73,13 @@ def load(
...
-        batch_size: how many instances to include in batches after loading
-        shuffle: whether or not to shuffle the dataset.  Defaults to True.
+        batch_size: (Optional) how many instances to include in batches after loading. If
+            not provided, no batching will occur.
+        shuffle: whether or not to shuffle the dataset.  Defaults to None, which indicates
+            shuffling will only occur if a `shuffle_buffer` is provided.
         shuffle_buffer: the size of the buffer to use in shuffling.
-        img_size: the size to resize the images to.
+        img_size: (Optional) size to resize the images to.  By default, images are not
+            resized and ragged batches are produced if batching occurs.
```
Comment (tanzhenyu) [inline keras_cv/datasets/pascal_voc/load.py:82]: "ragged batch" is confusing, consider "tf.RaggedTensor"

File: keras_cv/datasets/pascal_voc/load.py [hunk 2]
```diff
@@ -69,10 +73,13 @@ def load(
             For a list of supported formats, please  Refer
             [to the keras.io docs](https://keras.io/api/keras_cv/bounding_box/formats/)
             for more details on supported bounding box formats.
-        batch_size: how many instances to include in batches after loading
-        shuffle: whether or not to shuffle the dataset.  Defaults to True.
+        batch_size: (Optional) how many instances to include in batches after loading. If
+            not provided, no batching will occur.
+        shuffle: whether or not to shuffle the dataset.  Defaults to None, which indicates
```
Comment (tanzhenyu) [inline keras_cv/datasets/pascal_voc/load.py:78]: If that's the case, then we don't need a `shuffle` argument. Just `shuffle_buffer` being None would mean no shuffling, and being int would mean shuffling.

File: keras_cv/datasets/pascal_voc/load.py [hunk 3]
```diff
@@ -49,9 +53,9 @@ def load(
     split,
     bounding_box_format,
     batch_size=None,
-    shuffle=True,
+    shuffle=None,
     shuffle_buffer=None,
-    img_size=(512, 512),
+    img_size=None,
```
Comment (tanzhenyu) [inline keras_cv/datasets/pascal_voc/load.py:58]: IMHO, providing a `img_size` around `Resizing` is not a very useful argument, we might instead want to provide arguments around `RandomCropResize` (or whatever the name is)

**Review by tanzhenyu — COMMENTED** [review]
Thanks for the PR!

File: keras_cv/datasets/pascal_voc/load.py (hunk 3 above)
Comment (LukeWood) [reply keras_cv/datasets/pascal_voc/load.py:57]: To be honest, I'm considering just dropping the img_size argument altogether

Comment (LukeWood) [reply keras_cv/datasets/pascal_voc/load.py:57]: It feels like we mght want to just augment images indiviudally and use `tf.data.padded_batch`

Comment (LukeWood) [reply keras_cv/datasets/pascal_voc/load.py:58]: Although; it is nice for visualization.

Comment (tanzhenyu) [reply keras_cv/datasets/pascal_voc/load.py:58]: I think we should probably allow full customization for KPLs and probably don't even need a public data loader.

**Review by tanzhenyu — APPROVED** [review]
